src/scheduler/server_info.cpp

```diff
@@ -747,6 +749,9 @@ query_server_dyn_res(server_info *sinfo)
 				} else if (ret == 0) {
 					log_eventf(PBSEVENT_DEBUG, PBS_EVENTCLASS_SERVER, LOG_DEBUG, "server_dyn_res",
 					"Program %s timed out", conf.dynamic_res[i].command_line);
+				}
+				if (ret == -1 || ret == 0) {
```

Comment:
I'm not sure this is necessary, since it will get cleaned up in the lower check.

Reply:
Then why don't we just do it once at the end? Lean and mean...

Reply:
We can do several things. We can leave things as is, but that would entail reading a file descriptor tying us to a killed program in case of errors. That will work, but it looks weird to even attempt to read there. It _is_ the minimal change, though.

The second alternative is to just set pid to 0 when we've killed the child, set k to 0,  and surround the parent fdopen/fgets section with if (pid>0). In that case we'll just skip even trying to read something if we had an error on the select earlier i.e. if there's nothing to be read we shouldn't even try to read.

The third alternative is to just use "continue" to go on to the next server_dyn_res resource if we had an error on the select and we killed the child.

I don't have a preference. I'll defer to @vstumpf and @bhroam as to which os the variants they like best. I'll pick the first one because it is the minimally invasive one, but I may change that when you two review it and come to a consensus.

Reply:
I'm not sure we can use `continue` without also logging "Setting resource to 0" and calling `set_resource()`.
I like the second option.

Reply:
We could change the code flow a little and fix this.  We really have two code flows.  One error case where we kill the process and the normal case where we read/set the resource.  How about instead of checking for error first, we check for not error and do the read/set.  Either side of that if will drop us to the kill at the bottom.